### PR TITLE
Squelch spam

### LINKED ISSUE #2: Squelch spam
We should implement some kind of protection against message spamming (i.e., the same message repeated n times in x time period).

Perhaps squashing them and instead emitting a 'last message repeated n times' each x interval would suffice.

### PR DESCRIPTION
**Pull request by aremmell**
Closes #2 ... finally!

### CODE REVIEW

**Comment by johnsonjh** [comment]
I'll check it out and make sure it builds and works all the weird places and with all the compilers

**Comment by johnsonjh** [comment]
@aremmell so the TODO is trivial to merge, but, just FYI I decided against trying to say the FNV code is specifically public domain, it adds way more legalese than the couple lines of code and it's not at all required, so avoids another license file etc.  I think we might want to give a link to the FNV site in a comment though.

**Comment by johnsonjh** [comment]
OK that's done.  I wonder why Travis had failed before - Travis is being weird on my phone.

I'll check it back in the office in a bit ...

**Comment by johnsonjh** [comment]
@aremmell Looks like the failures are:

```
in 2.279sec

Failed tests:

	  • file-archive-large
	  • sanity-text-styles
	  • sanity-update-config
```

started failing with https://github.com/aremmell/libsir/pull/85/commits/0f62f8358e55449a5fc6984755b3f9fbe9e2ef14

It'll be a few hours before I can dive back in here.  Have you had a chance to check it out?

**Comment by aremmell** [comment]
> @aremmell Looks like the failures are:
> 
> ```
> in 2.279sec
> 
> Failed tests:
> 
> 	  • file-archive-large
> 	  • sanity-text-styles
> 	  • sanity-update-config
> ```
> 
> started failing with [0f62f83](https://github.com/aremmell/libsir/commit/0f62f8358e55449a5fc6984755b3f9fbe9e2ef14)
> 
> It'll be a few hours before I can dive back in here. Have you had a chance to check it out?

Hrmm, all the tests were passing on macOS. I'll look into it.

**Comment by aremmell** [comment]
> > @aremmell Looks like the failures are:
> > ```
> > in 2.279sec
> > 
> > Failed tests:
> > 
> > 	  • file-archive-large
> > 	  • sanity-text-styles
> > 	  • sanity-update-config
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > started failing with [0f62f83](https://github.com/aremmell/libsir/commit/0f62f8358e55449a5fc6984755b3f9fbe9e2ef14)
> > It'll be a few hours before I can dive back in here. Have you had a chance to check it out?
> 
> Hrmm, all the tests were passing on macOS. I'll look into it.

It's because I'm a moron and I forgot there's a few tests that repeat the same message over and over. 😂

I'll fix them up in a bit.

**Comment by aremmell** [comment]
Should be GTG now.

**Comment by johnsonjh** [comment]
I'm home but I've been slow cooking ribs (and drinking) for about 3 hours solid so my debugging confidence levels are somewhat degraded at the moment so thanks for looking out. :)

**Comment by aremmell** [comment]
> I'm home but I've been slow cooking ribs (and drinking) for about 3 hours solid so my debugging confidence levels are somewhat degraded at the moment so thanks for looking out. :)

I'm jealous of the drinking, but also not, because I can't stop once I start ;). Are you good to review this code? It's not terribly complex.

File: tests/tests.h [hunk 1]
```diff
@@ -230,10 +236,10 @@ bool sirtest_os_log(void);
 bool sirtest_filesystem(void);
 
 /**
- * @test Ensure an empty message is properly handled.
+ * @ test
  * @returns bool `true` if the test passed, `false` otherwise.
```
Comment (johnsonjh) [inline tests/tests.h:240]: IMG_7859

You'll want to add a test description here, I think.

**Review by johnsonjh — CHANGES_REQUESTED** [review]
Looks good other than the format string error, running autoformat.sh, and the missing test description.  

I haven't tested on everything yet, but I can do that ~tomorrow~ later today, and as this is a rather non-invasive update, I imagine any possible portability regressions introduced I could trivially fix up.

File: tests/tests.h (hunk 1 above)
Comment (johnsonjh) [reply tests/tests.h:240]: @aremmell Also, need to run autoformat.sh.  Then GTG

File: tests/tests.c [hunk 2]
```diff
@@ -1059,14 +1060,14 @@ bool sirtest_updatesanity(void) {
...
-        pass &= filter_error(sir_alert("modified config #%" PRIu32 "", rnd), SIR_E_NODEST);
-        pass &= filter_error(sir_emerg("modified config #%" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_debug("modified config #% (debug)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_info("modified config #% (info)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_notice("modified config #% (notice)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_warn("modified config #% (warning)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_error("modified config #% (error)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_crit("modified config #% (critical)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_alert("modified config #% (alert)" PRIu32 "", rnd), SIR_E_NODEST);
+        pass &= filter_error(sir_emerg("modified config #% (emergency)" PRIu32 "", rnd), SIR_E_NODEST);
```
Comment (johnsonjh) [inline tests/tests.c:1070]: This builds an invalid format string, breaks printing of "rnd"

> 04:39:29.986 [info] update_sanity(1816069): modified config #% (info)u

Need to rearrange it.

Comment (aremmell) [reply tests/tests.c:1070]: Good one, me.

**Comment by aremmell** [comment]
All done.

**Review by johnsonjh — APPROVED** [review]
👍🏻

**Comment by aremmell** [comment]
Wtf is the check 'queued' for? I shouldn't have any Travis builds running

**Pull request merged**